### ISSUE TITLE
Create Coin Search section - add to Portfolio & Watchlist

### ISSUE THREAD

**Issue by leongaban** [Futuratum/moonholdings.io#21]
The search should be redesigned to be larger to accommodate features such as adding to Portfolio and Watchlist, and displaying more information about coin selected.

**Design elements & functionality:**

- Search input
- Coin selection
- Coin information
- Add to Portfolio & Add to Watchlist
- Close X search button

---
**Tasks**

- [x] Refactor the Overlay component to stand on it's on and use Redux state
- [x] Display search results
- [x] Add endpoint `exchange-markets/prices` to get asset exchanges, then the price on that exchange
- [x] Design, create and style components _(80% complete)_
- [x] Wire up search functionality to API & Redux state
- [x] Disable Add buttons until Exchange has been selected
- [x] Wire up add to Portfolio `ADD_COIN`
- [x] Wire up add to Watch list `ADD_WATCH`
- [x] Wire in localStorage to save Portfolio & Watchlist

![screen shot 2019-02-19 at 11 50 30 pm](https://user-images.githubusercontent.com/191636/53069321-28c6b180-34a1-11e9-8999-174940e226d4.png)

**Comment by leongaban** [comment]
Bug when updating the Redux actions/reducers/store to accommodate new state for Overlay: https://stackoverflow.com/questions/54775023/error-when-using-combinereducers-error-reducer-assetsreducer-returned-undefi

**Comment by leongaban** [comment]
Fixed the previous Redux state bug, but now running into a Typescript error: https://stackoverflow.com/questions/54775648/running-into-typescript-state-type-mismatch-error-inside-of-redux-store-file

**Comment by leongaban** [comment]
Fixed both bugs above... now waiting on a response from Nomics about this question:
https://forums.nomics.com/t/question-what-is-the-best-way-to-get-all-the-supported-exchange-for-an-asset/117

Hopefully there is a way so we can include exchange selection per asset.

**Comment by leongaban** [comment]
@prochafilho just added you if you are curious, fixed some more advanced typescript issues. Also this is a gigantic task hehe

**Comment by leongaban** [comment]
New question regarding how to correct type the return response from the `getMarkets` async function:

https://stackoverflow.com/questions/54812453/function-lacks-ending-return-statement-and-return-type-does-not-include-undefin

**Comment by leongaban** [comment]
Trying to figure out the `dispatch` typing issue: https://stackoverflow.com/questions/54844839/typescript-how-to-type-the-dispatch-in-redux

**Comment by leongaban** [comment]
Completed and merged into `development`
